tensorflow_probability/python/experimental/bijectors/highway_flow.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import tensorflow as tf
+
+import tensorflow_probability as tfp
+
+tfd = tfp.distributions
+tfb = tfp.bijectors
+tfe = tf.experimental
+
+'''
+This initial implementation is based on the PyTorch implementation at
+https://github.com/LucaAmbrogioni/CascadingFlow/blob/main/modules/networks.py
+'''
+
+
+class TriResNet(tfb.Bijector):
+
+    # set activation=False for last block
+    def __init__(self, width, activation=True, validate_args=False, name='tri_res_net'):
+        super(TriResNet, self).__init__(
+            validate_args=validate_args,
+            forward_min_event_ndims=0,
+            name=name)
+
+        self.width = width
+        self.W = tf.Variable(np.random.normal(0, 0.01, (self.width, self.width)), trainable=True, dtype=tf.float32)
+        self.b = tf.Variable(np.random.normal(0, 0.01, (self.width,)), trainable=True, dtype=tf.float32)  # bias
+
+        # positive diagonal for U
+        self.d = tf.Variable(np.random.normal(0, 0.01, (self.width,)), trainable=True, dtype=tf.float32)
+
+        # lambda before sigmoid
+        self.pre_l = tf.Variable(np.random.normal(0, 0.1, (1,)), trainable=True, dtype=tf.float32)
+
+        self.activation = activation
+
+        self.masku = tfe.numpy.tril(tf.ones((self.width, self.width)), -1)
+        self.maskl = tfe.numpy.triu(tf.ones((self.width, self.width)), 1)
+
+    def get_l(self):
+        return tf.nn.sigmoid(self.pre_l)
+
+    def get_L(self):
+        return self.W * self.maskl + tf.eye(self.width)
+
+    def get_U(self):
+        return self.W * self.masku + tf.linalg.diag(tf.math.softplus(self.d))
+
+    def df(self, x):
+        # derivative of activation
+        l = self.get_l()
+        return l + (1 - l) * tf.math.sigmoid(x) * (1 - tf.math.sigmoid(x))
+
+    def cu(self, M):
```

Comment:
as above, we prefer descriptive, readable names: `convex_update` instead of `cu`, etc.

Also, most of these methods should probably be private (`_convex_update`, etc) unless you expect them to be used directly by external code.